mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeansBlocksImpl.scala

```diff
@@ -0,0 +1,397 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.mllib.clustering
+
+import scala.collection.mutable.ArrayBuffer
+
+import org.apache.spark.broadcast.Broadcast
+import org.apache.spark.internal.Logging
+import org.apache.spark.ml.feature.InstanceBlock
+import org.apache.spark.ml.util.Instrumentation
+import org.apache.spark.mllib.linalg.{BLAS, DenseMatrix, Vector, Vectors}
+import org.apache.spark.mllib.linalg.BLAS.axpy
+import org.apache.spark.rdd.RDD
+import org.apache.spark.storage.StorageLevel
+import org.apache.spark.util.Utils
+import org.apache.spark.util.random.XORShiftRandom
+
+/**
+ * K-means clustering using RDD[InstanceBlock] as input format.
+ */
+class KMeansBlocksImpl (
```

Comment:
We should not create a new public class in the `.mllib` side

Reply:
I think what we depend on the `.mllib` side when impling KMeans on blocks is just the initialization method, we can make it a method in the `.mllib` and use it in the `.ml` side.